src/homework/api/security.py

```diff
@@ -0,0 +1,127 @@
+import binascii
+import typing
+import enum
+import jwt
+import base64
+import json
+
+from sqlalchemy import create_engine, select
+from sqlalchemy.orm import Session
+from fastapi import Header, HTTPException
+
+from src.homework.api.contracts import AccessLevel
+from src.homework.db.engine import get_postgres_db_url
+from src.homework.db.models import Application
+
+
+class MasterAccessLevel(enum.Enum):
+    MASTER_APP = "MasterApp"
+
+
+BroadenAccessLevel = typing.Union[AccessLevel, MasterAccessLevel]
+
+
+def issue_token(
+    app_id: str, access_level: BroadenAccessLevel, secret: str
+) -> str:
+    """
+    Generate a token signed by secret.
+
+    :param app_id: str
+    :param access_level: BroadenAccessLevel
+    :param secret: str
+    :return: str
+    """
+    return jwt.encode(
+        {"app_id": app_id, "access_level": access_level.value},
+        secret,
+        algorithm="HS256",
+    )
+
+
+def verify_token(token: str, supposed_secret: str) -> typing.Dict[str, str]:
+    return jwt.decode(token, supposed_secret, algorithms="HS256")
+
+
+def get_app_by_id(
+    app_id: str, get_url_func: typing.Callable[[], str]
+) -> None | Application:
+    """
+    Get an application from the database by its id.
+
+    If the application with given id does not exist, function returns None.
+    :param app_id: str
+    :param get_url_func: a function to generate connection string
+    :return: None | Application
+    """
+    engine = create_engine(get_url_func())
+    with Session(engine) as session:
+        app = session.scalar(
+            select(Application).where(Application.app_id == app_id)
+        )
+        return app
+
+
+def decode_segment(segment: str) -> None | typing.Dict[str, str]:
```

Comment:
```suggestion
def decode_segment(segment: str) -> None | dict[str, str]:
```
This suggestion may enhance readability, although it is very minor

Reply:
Great idea, thanks! 
P. S. the idea works for Python3.9 and upper, which is suitable for us :)
https://stackoverflow.com/a/68908479